module/zfs/zio_compress.c

```diff
@@ -120,8 +117,24 @@ zio_compress_data(enum zio_compress c, abd_t *src, void *dst, size_t s_len)
 	if (c == ZIO_COMPRESS_EMPTY)
 		return (s_len);
 
-	/* Compress at least 12.5% */
-	d_len = s_len - (s_len >> 3);
+	/*
+	 * If data was compressed and passed ratio checks before -
+	 * we can skip these checks for reproducible results,
+	 * so set compress_threshold=0 for reproducibility of ARC checks.
+	 */
+	if (compress_threshold > 0) {
```

Comment:
I don't think we need the `if` statement.  If `compress_threshold` is zero, the code in this block will still do what we want.